src/clients/api.ts

```diff
@@ -1,239 +1,22 @@
-import {
-  ApiHeaderJson,
-  composeHeaderProducers,
-  composeResponseDecoders,
-  constantResponseDecoder,
-  createFetchRequestForApi,
-  ioResponseDecoder,
-  ReplaceRequestParams,
-  RequestHeaderProducer,
-  RequestParams,
-  TypeofApiCall
-} from "italia-ts-commons/lib/requests";
-import { ProblemJson } from "italia-ts-commons/lib/responses";
-import { Omit } from "italia-ts-commons/lib/types";
 import nodeFetch from "node-fetch";
-
-import {
-  createProfileDefaultDecoder,
-  CreateProfileT,
-  getMessageDefaultDecoder,
-  getMessagesByUserDefaultDecoder,
-  GetMessagesByUserT,
-  GetMessageT,
-  getProfileDefaultDecoder,
-  GetProfileT,
-  getServiceDefaultDecoder,
-  GetServiceT,
-  getUserDataProcessingDefaultDecoder,
-  GetUserDataProcessingT,
-  getVisibleServicesDefaultDecoder,
-  GetVisibleServicesT,
-  StartEmailValidationProcessT,
-  updateProfileDefaultDecoder,
-  UpdateProfileT,
-  upsertUserDataProcessingDefaultDecoder,
-  UpsertUserDataProcessingT
-} from "../../generated/io-api/requestTypes";
-
-// we want to authenticate against the platform APIs with the APIM header key or
-// the Azure Functions header key, so we send both headers
-function SubscriptionKeyHeaderProducer<P>(
-  token: string
-): RequestHeaderProducer<P, "X-Functions-Key" | "Ocp-Apim-Subscription-Key"> {
-  return () => ({
-    "Ocp-Apim-Subscription-Key": token,
-    "X-Functions-Key": token
-  });
-}
+import { Client, createClient } from "../../generated/io-api/client";
 
 export function APIClient(
   baseUrl: string,
   token: string,
   // tslint:disable-next-line:no-any
   fetchApi: typeof fetch = (nodeFetch as any) as typeof fetch // TODO: customize fetch with timeout
-): {
-  readonly updateProfile: TypeofApiCall<typeof updateProfileT>;
-  readonly getMessage: TypeofApiCall<typeof getMessageT>;
-  readonly getMessages: TypeofApiCall<typeof getMessagesT>;
-  readonly getProfile: TypeofApiCall<typeof getProfileT>;
-  readonly createProfile: TypeofApiCall<typeof createProfileT>;
-  readonly upsertUserDataProcessing: TypeofApiCall<
-    typeof upsertUserDataProcessingT
-  >;
-  readonly emailValidationProcess: TypeofApiCall<
-    typeof emailValidationProcessT
-  >;
-  readonly getService: TypeofApiCall<typeof getServiceT>;
-  readonly getVisibleServices: TypeofApiCall<typeof getVisibleServicesT>;
-  readonly getUserDataProcessing: TypeofApiCall<typeof getUserDataProcessingT>;
-} {
-  const options = {
+): Client<"SubscriptionKey"> {
+  return createClient<"SubscriptionKey">({
```

Comment:
this must be x-functions-key (Ocp-Apim-Subscription-Key is unused)

Reply:
It actually is. The [referenced api spec](https://github.com/pagopa/io-functions-app/blob/4f089cd50ade560d9c3a82bcb8b36c787e1cd59a/openapi/index.yaml#L347) is
```yaml
  SubscriptionKey:
    type: apiKey
    name: X-Functions-Key
    in: header
```

Under the hood, the generated client does the following:
```ts
headers: ({ SubscriptionKey }: { SubscriptionKey: string }) => ({
      "X-Functions-Key": `Bearer ${SubscriptionKey}`
    }),
```

Reply:
that's right, not easy to grasp from the code here :-) (maybe we could add a comment?)

Reply:
done [here](https://github.com/pagopa/io-backend/pull/711/commits/b2e7987c9698c47a08a65b85607030acba680d7f)